framework/src/modules/random/endpoint.ts

```diff
@@ -63,10 +63,36 @@ export class RandomEndpoint extends BaseEndpoint {
 		const seed = ctx.params.seed
 			? Buffer.from(ctx.params.seed, 'hex')
 			: cryptography.utils.generateHashOnionSeed();
-		const count = ctx.params.count ?? 1000000;
+		const count = ctx.params.count ?? MAX_HASH_COMPUTATION;
 		const distance = ctx.params.distance ?? 1000;
 
-		const hashes = cryptography.utils.hashOnion(seed, count, distance) as Buffer[];
+		if (distance > count) {
+			throw new Error('Invalid distance. Distance must be less than or equal to count.');
+		}
+
+		let hashes: Buffer[];
+		if (ctx.params.hashes) {
+			// when hashes is provided, count and distance must be provided together
+			if (!ctx.params.count || !ctx.params.distance) {
+				throw new Error('Hashes must be provided with count and distance.');
+			}
+			if (count % distance !== 0) {
+				throw new Error('Invalid count. Count must be multiple of distance.');
+			}
```

Comment:
Isn't this check relevant also when the hashes are not provided, and should go before `if (ctx.params.hashes)` check?

Reply:
yes, good catch 👍